src/audio/output.rs

```diff
@@ -0,0 +1,27 @@
+//! Provides structures and functions used to get audio outputs.
+
+use rodio::Endpoint;
+use rodio::*;
+
+/// A speaker(s) through which audio can be played.
+pub struct Output {
+    pub(crate) endpoint: Endpoint,
+}
+
+impl Output {
+
+    /// Gets the name of the output
+    pub fn get_name(&self) -> String {
+        self.endpoint.get_name()
+    }
+}
+
+/// Get the default output, returns none if no outputs are available.
+pub fn default_output() -> Option<Output> {
+    get_default_endpoint().map(|re| Output { endpoint: re })
+}
+
+/// Get a list of outputs available to the system.
+pub fn outputs() -> Vec<Output> {
+    get_endpoints_list().map(|re| Output {endpoint: re}).collect::<Vec<Output>>()
```

Comment:
I guess it isn't.  (I'm really looking forward to `impl Trait` so that I can have efficient interfaces that aren't a pain to write)